### PR TITLE
ARP: implement decoder and logger v8

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6827

Describe changes:
- Add events for unsupported hardware and protocol
- Remove `PrototypeToString` and `HwtypeToString` functions from logger
- Fix `dest_ip` length
- Add comments as requested

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1807

### CODE REVIEW

File: src/decode-arp.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if (!PacketIncreaseCheckLayers(p)) {
+        return TM_ECODE_FAILED;
+    }
+
+    const ARPHdr *arph = PacketSetARP(p, pkt);
+    if (unlikely(arph == NULL))
+        return TM_ECODE_FAILED;
+
+    if (ntohs(arph->hw_type) != ETHERNET_HW_TYPE) {
+        ENGINE_SET_INVALID_EVENT(p, ARP_INVALID_HARDWARE);
```
Comment (victorjulien) [inline src/decode-arp.c:49]: it's not invalid, is it? It's not supported by the decoder, but (possibly) still valid?

File: src/decode-arp.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return TM_ECODE_FAILED;
+    }
+
+    if (ntohs(arph->proto_type) != ETHERNET_TYPE_IP) {
+        ENGINE_SET_INVALID_EVENT(p, ARP_INVALID_PROTOCOL);
```
Comment (victorjulien) [inline src/decode-arp.c:54]: same question

File: src/decode-arp.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const ARPHdr *arph = PacketSetARP(p, pkt);
+    if (unlikely(arph == NULL))
+        return TM_ECODE_FAILED;
+
+    if (ntohs(arph->hw_type) != ETHERNET_HW_TYPE) {
+        ENGINE_SET_INVALID_EVENT(p, ARP_INVALID_HARDWARE);
+        return TM_ECODE_FAILED;
+    }
+
+    if (ntohs(arph->proto_type) != ETHERNET_TYPE_IP) {
```
Comment (victorjulien) [inline src/decode-arp.c:53]: please use `SCNtohs`

File: src/decode-arp.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    if (!PacketIncreaseCheckLayers(p)) {
+        return TM_ECODE_FAILED;
+    }
+
+    const ARPHdr *arph = PacketSetARP(p, pkt);
+    if (unlikely(arph == NULL))
+        return TM_ECODE_FAILED;
+
+    if (ntohs(arph->hw_type) != ETHERNET_HW_TYPE) {
```
Comment (victorjulien) [inline src/decode-arp.c:48]: `SCNtohs`

File: src/output-json.h [hunk 5]
```diff
@@ -50,6 +50,8 @@ typedef struct JsonAddrInfo_ {
     Port sp;
     Port dp;
     char proto[JSON_PROTO_LEN];
+    // With this flag, ports are logged only if they have meaningful values.
```
Comment (victorjulien) [inline src/output-json.h:53]: comment suggests the value of ports is evaluated, but this is not what I'm seeing?

**Review by victorjulien — CHANGES_REQUESTED** [review]
A few comments. Think we're almost ready.

File: src/decode-arp.c (hunk 1 above)
Comment (glongo) [reply src/decode-arp.c:49]: Yes, it's not invalid. I guess 'unsupported' would be more appropriate.

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/11040

**Pull request closed without merging**